### ISSUE TITLE
Save/Load Games with Pre-release on a Mac

### ISSUE THREAD

**Issue by ron-murhammer** [triplea-game/triplea#5945]
**How can the problem be recreated?**
Try loading this save game on a Mac: https://forums.triplea-game.org/topic/105/ai-development-discussion-and-feedback/233

**Comment by ron-murhammer** [comment]
@DanVanAtta @RoiEXLab Do one of you have a Mac to test save/load games in the pre-release?

**Comment by RoiEXLab** [comment]
I don't own a mac, but I have access to one at work.
I could try loading the savegame tomorrow but I don't have any dev environment there so I can really do much in terms of debugging

**Comment by ron-murhammer** [comment]
Ok. Mostly just want to see if we can reproduce it first so that would help.

**Comment by DanVanAtta** [comment]
No mac here, just linux.

**Comment by RoiEXLab** [comment]
@ron-murhammer What map is the savegame for?

I tried loading the savegame and nothing happened, no error message, no loaded savegame 🤔

**Comment by ron-murhammer** [comment]
Global 40 2nd. Can you try doing it on a windows or linux machine?

**Comment by RoiEXLab** [comment]
On master everything works fine, on my javafx branch (for #5917 ) I instantly ran into this error though:
```
Caused by: java.lang.IllegalStateException: friendlyUnits should but does not contain all units from targetsToPickFrom
```

On the mac at work I have like 1 or 2 maps installed, nothing too crazy, so that could be the reason it failed for me. But it doesn't explain why the error seemed to have been silent

**Comment by RoiEXLab** [comment]
I'll try some things in the coming days, but obviously I can't spent more than a couple of minutes at work on it

**Comment by DanVanAtta** [comment]
@RoiEXLab , if that is the same error message, @ron-murhammer pushed a patch for that bug somewhat recently. The patch  should be on current master.

**Comment by RoiEXLab** [comment]
FYI I'm still trying to reproduce this issue, but restrictive company policies (because of a recent cybersecurity incident) make this a non-trivial task.
1. Every TLS connection gets broken up so I had to manually import a certificate to the bundled JRE
2. This didn't solve the problem either because parts of "evil" github are blocked because it could contain malware, making it impossible to get to the map download window.

So it looks like I'll have to manually download the map at home and try to send it to myself somehow.
Could you point me to the right zip to download? It's a bit hard to find the correct map just from the yaml file :/

**Comment by ron-murhammer** [comment]
@RoiEXLab It should be this global 40 zip: https://github.com/triplea-maps/world_war_ii_global/releases/download/0.83/world_war_ii_global.zip

**Comment by RoiEXLab** [comment]
Ok @ron-murhammer @DanVanAtta I was finally able to test this scenario.
When starting triplea on my mac "normally" there's some weird stuff going on, some buttons don't seem to be working and the overall interface seems really unresponsive: When clicking the load savegame button, I get a file dialog where the files aren't correctly filtered and selecting a file does absolutely nothing, except closing the dialog. When launching it using the bundled JRE over the commandline I don't even get some sort of error message, it just silently fails.
Using a local jdk 11 installation (AdoptOpenJDK) I was able to load the savegame no problem and when starting I got into the game where the AI did its moves until I was allowed to do the American purchases. So it looks fine there.
But we need to figure out what could be causing this weirdness
